### PR TITLE
update to go 1.20 and dependencies

### PR DESCRIPTION
**Pull request by jlsherrill**
## Summary

Updates to go 1.20
Action to developers:
* Upgrade to go 1.20 (or 1.21) https://go.dev/doc/install
* Delete and redownload your golangci-lint binary:  
 `rm release/golangci-lint; make install-golangci-lint`
* if you are using GoLand, update to the latest ( 2023.2.1)  You may need to update multiple times to get to the latest.

## Testing steps
tests pass

### CODE REVIEW

File: pkg/seeds/seeds.go [hunk 1]
```diff
@@ -35,10 +35,6 @@ const (
 	batchSize = 500
 )
 
-func init() {
-	rand.Seed(time.Now().UnixNano())
-}
```
Comment (jlsherrill) [inline pkg/seeds/seeds.go:40]: this is deprecated;  https://pkg.go.dev/math/rand#Seed

File: pkg/pulp_client/status.go [hunk 2]
```diff
@@ -4,7 +4,7 @@ import zest "github.com/content-services/zest/release/v2023"
 
 func (r *pulpDaoImpl) Status() (*zest.StatusResponse, error) {
 	// Change this back to StatusRead(r.ctx) on next zest update
-	status, resp, err := r.client.StatusAPI.StatusRead1(r.ctx).Execute()
+	status, resp, err := r.client.StatusAPI.StatusRead(r.ctx).Execute()
```
Comment (jlsherrill) [inline pkg/pulp_client/status.go:7]: newer zest doesn't need this

**Review by rverdile — APPROVED** [review]

**Pull request merged**